graphql-database-manager-dynamo/src/main/java/com/fleetpin/graphql/database/manager/dynamo/DynamoDb.java

```diff
@@ -609,6 +609,54 @@ public int maxBatchSize() {
 
     }
 
+    @Override
+    public CompletableFuture<Boolean> destroyOrganisation(final String organisationId) {
+        final var organisationCondition = Condition.builder()
+                .comparisonOperator(ComparisonOperator.EQ)
+                .attributeValueList(AttributeValue.builder().s(organisationId).build())
+                .build();
+
+        final var associatedOrganisationItems = QueryRequest.builder()
+                .tableName(entityTable)
+                .keyConditions(Map.of("organisationId", organisationCondition))
+                .build();
+
+        final var deletedOrganisationFuture = new CompletableFuture<Boolean>();
+
+        client.query(associatedOrganisationItems)
+                .thenApply(response -> {
+                    if (!response.hasItems()) {
+                        deletedOrganisationFuture.complete(false);
+                        return new ArrayList<CompletableFuture<DeleteItemResponse>>();
+                    }
+
+                    return response.items()
+                            .stream()
+                            .map(item -> {
+                                final var deleteFoundItem = DeleteItemRequest.builder()
+                                        .tableName(entityTable)
+                                        .returnValues(ReturnValue.ALL_OLD)
+                                        .key(Map.of(
+                                                "organisationId", item.get("organisationId"),
+                                                "id", item.get("id")
+                                        ))
+                                        .build();
+
+                                return client.deleteItem(deleteFoundItem);
```

Comment:
think we should batch these. Will be crazy slow otherwise

Reply:
How should I go about batching this?

Now that you mention it, I could see this scaling poorly.

Reply:
yip BatchWriteItem. Would use a flatMap to group them into chunks or something on the stream.
Not quite sure how I would turn the stream into a stream of arrays 25 elements long

Reply:
I was thinking of using the Iterators#partition method for this, and partitioning by 25 items.

Reply:
cool be interested to see it